lib/_stream_readable.js

```diff
@@ -538,12 +538,12 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
   }
   // This is a brutally ugly hack to make sure that our error handler
   // is attached before any userland ones.  NEVER DO THIS.
-  if (!dest._events || !dest._events.error)
+  if (!dest._events || !dest._events.has('error'))
     dest.on('error', onerror);
-  else if (Array.isArray(dest._events.error))
-    dest._events.error.unshift(onerror);
+  else if (Array.isArray(dest._events.get('error')))
+    dest._events.get('error').unshift(onerror);
   else
-    dest._events.error = [onerror, dest._events.error];
+    dest._events.set('error', [onerror, dest._events.get('error')]);
```

Comment:
Last I checked npm (a few weeks ago) it was only about a dozen or so? I had actually thought about submitting PRs to those projects just so we could get the `.once()` optimizations.